File: py/Peripheral.py
# Peripheral.py
class Peripheral:
    def __init__(self, tipe="", tipeKoneksi="", fitur=None):
        self.tipe = tipe
        self.tipeKoneksi = tipeKoneksi
        self.fitur = fitur if fitur is not None else []
    
    def setTipe(self, tipe):
        self.tipe = tipe
    
    def setTipeKoneksi(self, tipeKoneksi):
        self.tipeKoneksi = tipeKoneksi
    
    def setFitur(self, fitur):
        self.fitur = fitur
    
    def getTipe(self):
        return self.tipe
    
    def getTipeKoneksi(self):
        return self.tipeKoneksi
    
    def getFitur(self):
        return self.fitur

File: cpp/Peripheral.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Peripheral
{
private:
    string tipe;
    string tipeKoneksi;
    vector<string> fitur; // Menggunakan vector untuk menyimpan daftar fitur

public:
    // Konstruktor default
    Peripheral() {}

    // Konstruktor dengan parameter
    Peripheral(string tipe, string tipeKoneksi, vector<string> fitur)
    {
        this->tipe = tipe;
        this->tipeKoneksi = tipeKoneksi;
        this->fitur = fitur;
    }

    // Setter untuk tipe
    void setTipe(string tipe)
    {
        this->tipe = tipe;
    }

    // Setter untuk tipeKoneksi
    void setTipeKoneksi(string tipeKoneksi)
    {
        this->tipeKoneksi = tipeKoneksi;
    }

    // Setter untuk fitur
    void setFitur(vector<string> fitur)
    {
        this->fitur = fitur;
    }

    // Getter untuk tipe
    string getTipe() const
    {
        return this->tipe;
    }

    // Getter untuk tipeKoneksi
    string getTipeKoneksi() const
    {
        return this->tipeKoneksi;
    }

    // Getter untuk fitur
    vector<string> getFitur() const
    {
        return this->fitur;
    }

    
    // Destruktor
    ~Peripheral() {}
};
